src/UndoController.js

```diff
@@ -0,0 +1,72 @@
+var diff = require('virtual-dom/diff');
+var patch = require('virtual-dom/patch');
+var virtualize = require('vdom-virtualize');
+var uuid = require('uuid');
+
+module.exports = class UndoController {
+
+  constructor(scribe, attrs){
+    //remove the default undo command as we will be replacing it
+    scribe.commands.undo.execute = function(){};
+    //keep a reference to this scribe instance as there can be multiple instances on the page
+    this.scribe = scribe;
+    //add an object to keep a list of our patches in reverse order
+    this.diffs = [];
+    //add a count to the number of undo commands
+    this.count = 0;
+    //set the last content so we have a point of comparison
+    this.lastContent = virtualize(this.scribe.el);
+    //add listeners
+    this.scribe.el.addEventListener('input', (e)=> this.onContentChanged(e));
+    this.scribe.el.addEventListener('keydown', (e)=> this.onKeyPressed(e));
+  }
+
+  onKeyPressed(e){
+    if(e.metaKey && e.keyCode === 90){
+      e.preventDefault();
+      //write the previous content into scribe's element
+      var diff = this.diffs.pop();
```

Comment:
@hmgibson23  will be limited later.

@theefer if you provide `undefined` to `patch` it fails silently so no need to worry.